v2/internal/testcommon/vcr/redact.go

```diff
@@ -109,6 +109,17 @@ func HideRecordingData(azureIDs creds.AzureIDs, s string) string {
 	return s
 }
 
+func HideRecordingDataWithCustomRedaction(azureIDs creds.AzureIDs, s string, redact map[string]string) string {
+	// Replace and hide all the custom data
+	if redact != nil {
```

Comment:
It looks guaranteed to me now that redact is not nil? Even if it is nil, it's OK to `range` over a nil map it shouldn't panic right?